pages/faucet/index.tsx

```diff
@@ -0,0 +1,168 @@
+import { GetServerSideProps, InferGetServerSidePropsType, NextPage } from 'next'
+import Head from 'next/head'
+import React, { useCallback, useState } from 'react'
+import {
+  GoogleReCaptchaProvider,
+  useGoogleReCaptcha,
+} from 'react-google-recaptcha-v3'
+import { getAddress } from 'viem'
+
+import { useTransactionContext } from '../../contexts/transaction-context'
+import { Asset } from '../../model/asset'
+import { fetchAssets } from '../../api/asset'
+
+export const FAUCET_AMOUNTS: { [symbol: string]: number } = {
+  ETH: 0.1,
+  USDC: 100,
+  DAI: 100,
+  USDT: 100,
+  WBTC: 0.1,
+}
+
+const FaucetForm = ({ assets }: { assets: Asset[] }) => {
+  const { executeRecaptcha } = useGoogleReCaptcha()
+  const { setConfirmation } = useTransactionContext()
+  const [address, setAddress] = useState('')
```

Comment:
What if we used `useAccount` instead of getting a text input from the user. It's less prone to abuse and the user doesn't have to type in his address.

Reply:
https://github.com/clober-dex/coupon.finance/pull/33/commits/df941b26f5a12d400b021b37dfb7789895e3ad75 @onemonster Good point!